recipes/protobuf-c/all/conanfile.py

```diff
@@ -0,0 +1,104 @@
+        copy(self, "protobuf-c.cmake", src=self.recipe_folder, dst=self.export_sources_folder, keep_path=False)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def build_requirements(self):
+        # Since the package using protobuf-c will also need to use protoc (part of protobuf),
+        # we want to make sure the protobuf dep is visible, but the visible param is only available in v2
+        # TODO: Remove after dropping Conan 1.x
+        if conan_version >= Version("2"):
+            self.tool_requires("protobuf/3.21.9", visible=True)
+        else:
+            self.tool_requires("protobuf/3.21.9")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.9")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_PROTO3"] = self.options.with_proto3
+        tc.cache_variables["BUILD_PROTOC"] = self.options.with_protoc
+        # FIXME: Add shared library support. It was just too hairy to figure out initially
+        # tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        tc.cache_variables["BUILD_TESTS"] = False
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure(build_script_folder=os.path.join(self.source_folder, "build-cmake"))
+        cmake.build()
+
+    @property
+    def _cmake_install_base_path(self):
+        return os.path.join("lib", "cmake", "protobuf-c")
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        # TODO: This won't be needed once upstream PR (https://github.com/protobuf-c/protobuf-c/pull/555) gets merged
+        copy(self, "protobuf-c.cmake", dst=os.path.join(self.package_folder, self._cmake_install_base_path), src=self.export_sources_folder)
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        # upstream CMake config file name and target name matches the package name
+        self.cpp_info.libs = ["protobuf-c"]
+        self.cpp_info.builddirs.append(self._cmake_install_base_path)
+        # TODO: This won't be needed once upstream PR (https://github.com/protobuf-c/protobuf-c/pull/555) gets merged
+        self.cpp_info.set_property("cmake_build_modules", [
+            os.path.join(self._cmake_install_base_path, "protobuf-c.cmake")
+        ])
+        self.buildenv_info.append_path("PATH", os.path.join(self.package_folder, "bin"))
```

Comment:
```suggestion
```
For Conan 2.x both buildenv_info and runenv_info are automatically added to PATH.

Reply:
@uilianries, if I remove this line I get the following error:
```
CMake Error at /Users/gegles/.conan2/p/b/proto04030cf11be22/p/lib/cmake/protobuf-c/protobuf-c.cmake:2 (find_program):
  Could not find PROTOC_GEN_C using the following names: protoc-gen-c
Call Stack (most recent call first):
  build/apple-clang-14-x86_64-17-release/generators/protobuf-c-config.cmake:37 (include)
  CMakeLists.txt:4 (find_package)
```